File: Difficulty: Easy/Floor in a Sorted Array/floor-in-a-sorted-array.js
//{ Driver Code Starts
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => { inputString += inputStdin; });

process.stdin.on('end', _ => {
    inputString =
        inputString.trim().split('\n').map(string => { return string.trim(); });

    main();
});

function readLine() { return inputString[currentLine++]; }

function main() {
    let tc = parseInt(readLine());
    while (tc > 0) {
        let arr = readLine().split(' ').map(Number);
        let k = parseInt(readLine());

        let obj = new Solution();
        let res = obj.findFloor(arr, k);

        console.log(res); // Print the array as a space-separated string
        console.log("~");
        tc--;
    }
}

// } Driver Code Ends


// User function Template for javascript

/**
 * @param {number[]} arr
 * @param {number} k
 * @returns {number}
 */

class Solution {

    findFloor(arr, k) {
        // your code here
        for(let i=0;i<arr.length;i++){
            if(i>=0 && arr[i] >= k){
                if(arr[i] == k) return i;
                return i-1;
            }
        }
        return -1;
    }
}

File: Difficulty: Easy/Floor in a Sorted Array/floor-in-a-sorted-array.py
class Solution:
    #User function Template for python3
    
    #Complete this function
    def findFloor(self, arr, x):
        #Your code here
        index = -1
        currentMax = -999999
        for i in range(len(arr)):
            if arr[i] <= x and arr[i] >= currentMax:
                currentMax = arr[i]
                index = i
                
        return index


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        arr = list(map(int, input().strip().split()))
        x = int(input().strip())
        ob = Solution()
        ans = ob.findFloor(arr, x)
        print(ans)
        tc -= 1

# } Driver Code Ends